crates/shared/src/price_estimation/trade_verifier.rs

```diff
@@ -400,16 +414,86 @@ fn encode_settlement(
         &query.in_amount.get(),
     );
 
-    EncodedSettlement {
+    let mut trades = vec![encoded_trade];
+
+    for jit_order in trade.jit_orders.iter() {
+        let order_data = OrderData {
+            sell_token: jit_order.sell_token,
+            buy_token: jit_order.buy_token,
+            receiver: Some(jit_order.receiver),
+            sell_amount: jit_order.sell_amount,
+            buy_amount: jit_order.buy_amount,
+            valid_to: jit_order.valid_to,
+            app_data: AppDataHash::from_str(&jit_order.app_data)?,
+            fee_amount: 0.into(),
+            kind: match &jit_order.side {
+                dto::Side::Buy => OrderKind::Buy,
+                dto::Side::Sell => OrderKind::Sell,
+            },
+            partially_fillable: false,
```

Comment:
This is a slightly problematic detail I didn't consider before. The JIT order API for the `/settle` call doesn't allow for partially fillable orders because that assumes only truly new JIT orders being proposed there and for those fill-or-kill should be sufficient.
But for the `/quote` endpoint the JIT orders returned could also be user orders (masked as JIT orders to not introduce even more enums). So if a solver would like to return a quote based on a partially fillable user order in the book this would not work here.
I think the API needs to allow sending the `partially_fillable` flag with the JIT orders.

Reply:
That means, the solvers API needs to be also changed since now it doesn't contain this field in the JitOrder json:
https://github.com/cowprotocol/services/blob/39b3f5bda1d3909d0e07aa0f0fe3f8a460a014f2/crates/driver/src/infra/solver/dto/solution.rs#L278-L298

Reply:
It was intended like this in the `/solve` API but I guess it makes sense to keep these 2 APIs the same.